### PR TITLE
adding rules for vm detection through wmi calls to drive and motherbo…

### PR DESCRIPTION
**Pull request by anders-v**
…ard model

### CODE REVIEW

File: anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# generated using capa explorer for IDA Pro
+rule:
+  meta:
+    name: reference anti-VM strings - WMI disk drive model contains Virtual
```
Comment (williballenthin) [inline anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml:4]: perhaps: "detect VM via WMI disk drive queries"?

File: anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-motherboard-model-contains-virtual.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# generated using capa explorer for IDA Pro
+rule:
+  meta:
+    name: reference anti-VM strings - WMI motherboard model contains Virtual
```
Comment (williballenthin) [inline anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-motherboard-model-contains-virtual.yml:4]: perhaps: "detect VM via hardware motherboard WMI queries"?

File: anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    namespace: anti-analysis/anti-vm/vm-detection
+    authors:
+      - anders.vejlby@mandiant.com
+    scope: function
+    att&ck:
+      - Defense Evasion::Virtualization/Sandbox Evasion::System Checks [T1497.001]
+    references:
+      - <insert_references>
```
Comment (williballenthin) [inline anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml:12]:
```suggestion
```

File: anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    examples:
+      - E69F2E964AA4A4E676F3335705B047E7:0x4035e0
+  features:
+    - or:
+      - string: "Win32_DiskDrive"
+      - string: "Model"
+      - string: "Virtual"
```
Comment (williballenthin) [inline anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml:19]: is it reasonable to also include logic around WMI query execution? or is that code found in a different function?

File: anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    namespace: anti-analysis/anti-vm/vm-detection
+    authors:
+      - anders.vejlby@mandiant.com
+    scope: function
+    att&ck:
+      - Defense Evasion::Virtualization/Sandbox Evasion::System Checks [T1497.001]
+    references:
+      - <insert_references>
+    examples:
+      - E69F2E964AA4A4E676F3335705B047E7:0x4035e0
```
Comment (williballenthin) [inline anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml:14]: this file is on VT but its huge (18MB). maybe we should look for another candidate or move the rule to the nursery.

File: anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-motherboard-model-contains-virtual.yml (hunk 2 above)
Comment (anders-v) [reply anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-motherboard-model-contains-virtual.yml:4]: I am good with changing the title to this, was just trying to reuse the format of the other ones that reference strings.

File: anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml (hunk 1 above)
Comment (anders-v) [reply anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml:4]: I am good with changing the title to this, was just trying to reuse the format of the other ones that reference strings.

File: anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  features:
+    - or:
```
Comment (williballenthin) [inline anti-analysis/anti-vm/vm-detection/reference-anti-vm-strings---wmi-disk-drive-model-contains-virtual.yml:16]:
```suggestion
    - and:
```

**Review by mr-tz — COMMENTED** [review]
How about we combine the rules into `detect VM via hardware WMI query` and `or` the `Win32_*` strings?

**Comment by anders-v** [comment]
File should be added in a PR to capa-testfiles.

Den ons. 31. maj 2023 kl. 17.02 skrev Willi Ballenthin <
***@***.***>:

**Review by williballenthin — APPROVED** [review]

**Comment by mr-tz** [comment]
rules with FAIL:
  - covertly decode and write data to Windows directory using indirect calls
  - detect VM via disk hardware WMI queries
  - detect VM via motherboard hardware WMI queries

**Pull request merged**